panoramapublic/test/src/org/labkey/test/components/panoramapublic/PanoramaPublicSearchWebPart.java

```diff
@@ -0,0 +1,154 @@
+
+    public PanoramaPublicSearchWebPart(WebDriver driver, String title)
+    {
+        super(driver, title);
+    }
+
+    @Override
+    protected PanoramaPublicSearchWebPart.ElementCache newElementCache()
+    {
+        return new PanoramaPublicSearchWebPart.ElementCache();
+    }
+
+    public String getTitle()
+    {
+        return elementCache().title.get();
+    }
+
+    public PanoramaPublicSearchWebPart setTitle(String title)
+    {
+        elementCache().title.set(title);
+        return this;
+    }
+
+    public String getAuthors()
+    {
+        return elementCache().authors.get();
+    }
+
+    public PanoramaPublicSearchWebPart setAuthors(String value)
+    {
+        elementCache().authors.set(value);
+        return this;
+    }
+
+    public String getOrganism()
+    {
+        return elementCache().organism.get();
+    }
+
+    public PanoramaPublicSearchWebPart setOrganism(String value)
+    {
+        elementCache().organism.set(value);
+        return this;
+    }
+
+    public String getInstrument()
+    {
+        return elementCache().instrument.get();
+    }
+
+    public PanoramaPublicSearchWebPart setInstrument(String value)
+    {
+        elementCache().instrument.set(value);
+        return this;
+    }
+
+    public String getProtein()
+    {
+        return elementCache().protein.get();
+    }
+
+    public PanoramaPublicSearchWebPart setProtein(String value)
+    {
+        elementCache().protein.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart setProteinExactMatch(boolean value)
+    {
+        elementCache().proteinExactMatch.set(value);
+        return this;
+    }
+
+    public String getPeptide()
+    {
+        return elementCache().peptide.get();
+    }
+
+    public PanoramaPublicSearchWebPart setPeptide(String value)
+    {
+        elementCache().peptide.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart setPeptideExactMatch(boolean value)
+    {
+        elementCache().peptideExactMatch.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoExperimentSearch()
+    {
+        getWrapper().click(Locator.tagWithText("span", "Experiment Search"));
+        getWrapper().waitForElementToBeVisible(Locator.name("Authors"));
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoProteinSearch()
+    {
+        elementCache().proteinTab.click();
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoPeptideSearch()
+    {
+        elementCache().peptideTab.click();
+        return this;
+    }
+
+    public DataRegionTable search()
+    {
+        elementCache().search.click();
+        return new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).find(this);
+    }
+
+    public DataRegionTable refreshedTable()
+    {
+        return new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).find(this);
+    }
```

Comment:
Should wait for search results grid to update. Assuming the grid doesn't appear until you search once; probably something like:
```suggestion
    public DataRegionTable search()
    {
    
        Optional<DataRegionTable> resultsGrid = new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).findOptional(this);
        if (resultsGrid.isPresent())
        {
            resultsGrid.get().doAndWaitForUpdate(() -> elementCache().search.click());
            return resultsGrid.get();
        }
        else
        {
            elementCache().search.click();
            return findResultsGrid();
        }
    }

    public DataRegionTable findResultsGrid()
    {
        return new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).waitFor(this);
    }
```

Reply:
Done

Reply:
After doing the above change, test started failing waiting for cssSignal for doAndWaitForUpdate